### DISCUSSION TITLE
Toogle windows state works curiously

### DISCUSSION THREAD

**Discussion by luriusTM** [eythaann/Seelen-UI#205] [Q&A]
Hi, 

This is a question at first, because I do not get the reasons behind. For me, the "async_force_set_foreground" does not work as intended. Most of the time, when I try to switch window on taskbar, then my item not get focused, it got minimised. I digged it a bit, and found ,that the logical work in handler seems OK, but the method when I want to focus, calls "WindowsApi::async_force_set_foreground(hwnd)". This method in its content calls: 
![image](https://github.com/user-attachments/assets/aba9500c-7871-41b4-9b84-8215a70999c7)

The problem with this, this API enforce the state of minimised, and when it got the restore, it does not affect the window. If I replace this method with "SW_MINIMIZE", then it works perfectly, only it flashes and makes me crazy... 

When I replaced the "async_force_set_foreground" with the real manner in its core, it works perfectly, without any flashes.
![image](https://github.com/user-attachments/assets/75c4f4a1-1904-4e8d-8bc8-d32a209f97b6)

Do you have a specific reason why you are use that exact method?

**Comment by eythaann** [comment]
set_foreground doesn't work at all, windows doesn't allow other apps steal focus, btw I will check how improve this later.

**Comment by luriusTM** [comment]
Since then, from testing it turned out "bring_to" cause apps to be "AlwaysOnTop" features like, so it does not help, but the set_Foreground works form me like a charm. I stash it, until I dig more in the topic...
